src/basic/time-util.c

```diff
@@ -185,7 +185,7 @@ usec_t triple_timestamp_by_clock(triple_timestamp *ts, clockid_t clock) {
 usec_t timespec_load(const struct timespec *ts) {
         assert(ts);
 
-        if (ts->tv_sec == (time_t) -1 && ts->tv_nsec == (long) -1)
+        if (ts->tv_sec < 0 && ts->tv_nsec < 0)
```

Comment:
Hm, what does it mean to have e.g. tv_sec == -1, tv_nsec == 100? Shouldn't this be `||` instead?